app/controllers/cuisine_controller.rb

```diff
@@ -0,0 +1,69 @@
+class CuisineController < ApplicationController
+
+  before_action :find_related_restaurants, :only => [:delete, :destroy]
```

Comment:
how about a `before_action :set_cuisine`
with: `@cuisine = Cuisine.find(params.require(:id))` 

instead of setting it in every action.

Reply:
You're right. Fixed in restaurantsController as well.